components/org.wso2.carbon.privacy.forgetme.analytics.streams/src/main/java/org/wso2/carbon/privacy/forgetme/analytics/streams/instructions/AnalyticsStreamsInstructionReader.java

```diff
@@ -0,0 +1,120 @@
+import com.google.gson.JsonParseException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.wso2.carbon.privacy.forgetme.analytics.streams.beans.Streams;
+import org.wso2.carbon.privacy.forgetme.analytics.streams.exceptions.AnalyticsStreamsProcessorException;
+import org.wso2.carbon.privacy.forgetme.api.runtime.Environment;
+import org.wso2.carbon.privacy.forgetme.api.runtime.ForgetMeInstruction;
+import org.wso2.carbon.privacy.forgetme.api.runtime.InstructionReader;
+import org.wso2.carbon.privacy.forgetme.api.runtime.ModuleException;
+
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Properties;
+
+/**
+ * Implements instruction generation for analytics persisted streams.
+ */
+public class AnalyticsStreamsInstructionReader implements InstructionReader {
+
+    private static final Logger logger = LoggerFactory.getLogger(AnalyticsStreamsInstructionReader.class);
+
+    /**
+     * Check if the file is a JSON file.
+     *
+     * @param file File
+     * @return
+     */
+    private static boolean isJsonFile(File file) {
+        int index = file.getName().lastIndexOf('.');
+        return index > 0 && file.getName().substring(index + 1).equalsIgnoreCase("json");
+    }
+
+    @Override
+    public String getType() {
+        return "analytics-streams";
+    }
+
+    @Override
+    public List<ForgetMeInstruction> read(Path contentDirectory, Properties properties, Environment environment)
+            throws ModuleException {
+        List<ForgetMeInstruction> instructions = new ArrayList<>();
+
+        // Get all the JSON files and build the stream object model. The model is required to pass into the instruction.
+        File dir = contentDirectory.toFile();
+        File[] files = dir.listFiles(file -> file.isFile() && isJsonFile(file));
+        if (files != null) {
+            List<Streams.Stream> streams = loadStreamDefinitions(files);
+            instructions.add(new AnalyticsStreamsInstruction(environment, streams));
+        } else {
+            throw new AnalyticsStreamsProcessorException("No stream definition files found.");
+        }
+        return instructions;
+    }
+
+    /**
+     * Read all the stream definitions in the file system.
+     *
+     * @param streamFiles List of files which contains stream definitions
+     * @return
+     */
+    private List<Streams.Stream> loadStreamDefinitions(File[] streamFiles) {
+        List<Streams.Stream> streams = new ArrayList<>();
+        for (File streamFile : streamFiles) {
+            try {
+                Streams streamCollection = readStreamDefinitionFile(streamFile);
+                streams.addAll(streamCollection.getStreams());
+            } catch (AnalyticsStreamsProcessorException e) {
+                logger.error("Error in parsing the stream definition. Ignoring and passing to next file...", e);
```

Comment:
Don't we need to break the flow here on exception ?

Reply:
No. Since we are reading file-by-file here, inability to read a single file should not break the entire flow.